pkg/R/DataFrame.R

```diff
@@ -663,6 +663,33 @@ setMethod("toRDD",
             })
           })
 
+#' GroupBy
+#'
+#' Groups the DataFrame using the specified columns, so we can run aggregation on them.
+#'
+setGeneric("groupBy", function(x, ...) { standardGeneric("groupBy") })
+
+setMethod("groupBy",
+           signature(x = "DataFrame"),
+           function(x, ...) {
+             cols <- list(...)
+             if (length(cols) >= 1 && class(cols[[1]]) == "character") {
+               sgd <- callJMethod(x@sdf, "groupBy", cols[[1]], listToSeq(cols[-1]))
+             } else {
+               jcol <- lapply(cols, function(c) { c@jc })
+               sgd <- callJMethod(x@sdf, "groupBy", listToSeq(jcol))
+             }
+             groupedData(sgd)
+           })
+
+
+setMethod("agg",
```

Comment:
I think `agg` needs a generic here? I don't see it in existing `NAMESPACE` or as a method of base R